src/ORM/Behavior/EnumBehavior.php

```diff
@@ -0,0 +1,160 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\ORM\Behavior;
+
+use ArrayObject;
+use BackedEnum;
+use Cake\Datasource\EntityInterface;
+use Cake\Event\EventInterface;
+use Cake\ORM\Behavior;
+use Cake\ORM\Entity;
+use Cake\ORM\Query;
+
+/**
+ * Enum behavior
+ *
+ * Allows you to map a specific field in your model to a backed enum
+ *
+ *  * Add behavior + config to connect a field with a backed enum
+ * ```
+ * $this->addBehavior('Enum', [
+ *     'fieldMap' => [
+ *         'published' => ArticleStatus::class
+ *     ]
+ * ]);
+ * ```
+ */
+class EnumBehavior extends Behavior
+{
+    /**
+     * Default config
+     *
+     * These are merged with user-provided configuration when the behavior is used.
+     *
+     * @var array<string, mixed>
+     */
+    protected array $_defaultConfig = [
+        'fieldMap' => [],
+    ];
+
+    /**
+     * Initialize hook
+     *
+     * @param array<string, mixed> $config The config for this behavior.
+     * @return void
+     */
+    public function initialize(array $config): void
+    {
+        if (isset($config['fieldMap'])) {
+            $this->setConfig('fieldMap', $config['fieldMap']);
+        }
+    }
+
+    /**
+     * Transform array data (which may contain enum instances) into entity data
+     *
+     * @param \Cake\Event\EventInterface $event The beforeMarshal event that was fired
+     * @param \ArrayObject $data Data present before converting to an entity
+     * @param \ArrayObject $options Options passed to the event
+     * @return void
+     */
+    public function beforeMarshal(EventInterface $event, ArrayObject $data, ArrayObject $options): void
+    {
+        $fieldMap = $this->getConfig('fieldMap');
+        /** @var \BackedEnum $enumType */
+        foreach ($fieldMap as $field => $enumType) {
+            if ($data->offsetExists($field)) {
+                $value = $data->offsetGet($field);
+                if ($value instanceof BackedEnum) {
+                    $value = $value->value;
```

Comment:
Shouldn't marshaling be converting string values to enum instances instead of the other way round?

Reply:
If I don't keep this in there I can't do
```
$this->articles->newEntity([
    'published' => ArticleStatus::PUBLISHED,
]);
```
because I get a `Object of class TestApp\Model\Enum\ArticleStatus could not be converted to string`

Reply:
I believe that error is coming from `StringType::toDatabase()`, so we need to update it to check if the value is an enum and if so use the enum's value.

Reply:
This was resolved via a discussion in discord